quic/s2n-quic-platform/src/io/turmoil.rs

```diff
@@ -80,26 +69,50 @@ impl Io {
             ));
         };
 
-        let rx_buffer = buffer::Buffer::new_with_mtu(max_mtu.into());
-        let mut rx = socket::Queue::new(rx_buffer);
+        let local_addr = socket.local_addr()?;
+        let local_addr: inet::SocketAddress = local_addr.into();
+        let payload_len: usize = max_mtu.into();
+        let payload_len = payload_len as u32;
+
+        let (rx, rx_producer) = {
+            let entries = 1024;
 
-        let tx_buffer = buffer::Buffer::new_with_mtu(max_mtu.into());
-        let tx = socket::Queue::new(tx_buffer);
+            let mut consumers = vec![];
 
-        let local_addr: SocketAddress = socket.local_addr()?.into();
+            let (producer, consumer) = ring::pair(entries, payload_len);
+            consumers.push(consumer);
 
-        // tell the queue the local address so it can fill it in on each message
-        rx.set_local_address(local_addr.into());
+            let rx = rx::Rx::new(consumers, max_mtu, local_addr.into());
 
-        let instance = Instance {
+            (rx, producer)
+        };
+
+        let (tx, tx_consumer) = {
+            let entries = 1024;
```

Comment:
nit: I'm assuming 1024 is just arbitrary size that fits a reasonable amount of message, but might be good to call that out in a comment.